### PR TITLE
Rework CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by svrnm**
I started to write down some additional details to our contributors guidelines, I also used the opportunity to add some generic pieces.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,14 +1,118 @@
...
+[OpenTelemetry.io](https://opentelemetry.io/)! Here are a few general guidelines
+on contributing and reporting bugs that we ask you to review. Following these
+guidelines helps to communicate that you respect the time of the contributors
+managing and developing this open source project. In return, they should
+reciprocate that respect in addressing your issue, assessing changes, and
+helping you finalize your pull requests. In that spirit of mutual respect, we
+endeavor to review incoming issues and pull requests, and will close any
+lingering issues or pull requests after long times of inactivity.
+
+Please note that all of your interactions in the project are subject to our
```
Comment (chalin) [inline CONTRIBUTING.md:13]:
```suggestion
Note that all of your interactions in the project are subject to our
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,14 +1,118 @@
...
-### Cloud-IDE setup
+To help us and yourself to get your PR merged, please follow these practices:
+
+- Work from a fork: Click the
+  [Fork](https://github.com/open-telemetry/opentelemetry.io/fork) button at the
+  top of the repository and clone the fork locally. When you are ready, raise a
+  PR with the upstream repository.
+- Do not work from the `main` branch of your fork, but create a PR-specific
+  branch.
+- Make sure that maintainers are
```
Comment (chalin) [inline CONTRIBUTING.md:90]:
```suggestion
- Ensure that maintainers are
```

File: .vscode/cspell.json [hunk 3]
```diff
@@ -74,6 +75,7 @@
     "mongodb",
     "mysql",
     "ndjson",
+    "netlify",
```
Comment (chalin) [inline .vscode/cspell.json:78]: Please also add `Netlify` to `.textlintrc`.

Comment (chalin) [reply .vscode/cspell.json:78]: We should also add a rule for GitPod, but I can do that after this is merged if you prefer (we'll need to allow for gitpod.io).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,14 +1,118 @@
...
+For everything else, follow the
+[instructions to setup a development environment](#development) below.
 
-## Dev setup
+### PR Guidelines
 
-### Cloud-IDE setup
+To help us and yourself to get your PR merged, please follow these practices:
+
+- Work from a fork: Click the
```
[the remark this answers is not in the document]
Comment (svrnm) [reply CONTRIBUTING.md:84]: Even a quick fix requires a fork if you do not have permissions to create a branch on upstream. We are the exception to that.

[the remark this answers is not in the document]
Comment (chalin) [reply CONTRIBUTING.md:84]: Oh right!

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,14 +1,118 @@
...
+reported or fixed by searching through our
+[issues list](https://github.com/open-telemetry/opentelemetry.io/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc).
+
+When creating a new issue, please be sure to include a title and clear
+description, as much relevant information as possible, and, if possible, a test
+case.
+
+If you discover a security issue, please do not report it through GitHub.
+Instead, please see security procedures in our
+[Security Policy](https://github.com/open-telemetry/opentelemetry.io/security/policy).
```
[the remark this answers is not in the document]
Comment (svrnm) [reply CONTRIBUTING.md]: done

File: .vscode/cspell.json (hunk 3 above)
Comment (svrnm) [reply .vscode/cspell.json:78]: Done, I also added GitPod, didn't have any issues with the URLs

File: .textlintrc.yml [hunk 6]
```diff
@@ -36,6 +36,7 @@ rules:
       - Erlang
       - GCP
       - GitHub
+      - GitPod
```
Comment (chalin) [inline .textlintrc.yml:39]:
```suggestion
      - Gitpod
```

**Review by chalin — APPROVED** [review]
LGTM aside from one comment.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -23,11 +131,11 @@ cloud IDE:
     > repository, or just want to look around, open
     > <https://gitpod.io/#https://github.com/open-telemetry/opentelemetry.io>.
 
-Gitpod will automatically install the repo-specific packages for you. You're now
+GitPod will automatically install the repo-specific packages for you. You're now
```
Comment (svrnm) [inline CONTRIBUTING.md:134]:
```suggestion
Gitpod will automatically install the repo-specific packages for you. You're now
```

**Review by cartermp — APPROVED** [review]
LGTM. I think the preamble's a little wordy, but it's still good.

**Pull request merged**